src/Processors/Transforms/JoiningTransform.cpp

```diff
@@ -266,7 +267,9 @@ Blocks JoiningTransform::readExecute(Chunk & chunk)
 FillingRightJoinSideTransform::FillingRightJoinSideTransform(Block input_header, JoinPtr join_)
     : IProcessor({input_header}, {Block()})
     , join(std::move(join_))
-{}
+{
+    spillable = typeid_cast<GraceHashJoin *>(join.get());
+}
```

Comment:
Generally speaking, a `grace_hash` spill can also be triggered by the `DelayedJoinedBlocksWorkerTransform` when processing any of the n-th buckets, particularly in cases of data distribution skew.
We could either add support for spilling while processing subsequent buckets or proceed with the current, simpler implementation and revisit the it later if needed.

Reply:
There can also be multiple `FillingRightJoinSideTransform` instances for a single join. Do we account for the memory occupied by one join, as well as cases where several joins exist within a query (so, some processors report about the same data in memory) ?

Reply:
We could make `DelayedJoinedBlocksWorkerTransform` spillable as need later.

I have thought about the case that some processors will share the same data. That is how I consider
1. The spill execution should be executed by one thread at the same time, if it's shared among processors. I think most of them have been designed in this way.
2. `spillOnSize` needs a argument `bytes`, it indicates that when the processor is triggered to spill how large the data is. When a procssor `p1` enters `spillOnSize` and the data has been spilled by another processor `p2`. `p1` will not take any action since the current size of data is less then `bytes`.